pkg/tfvars/openstack/openstack.go

```diff
@@ -101,24 +105,15 @@ func TFVars(
 		userManagedLoadBalancer = true
 	}
 
-	var zones []string
-	{
-		seen := make(map[string]struct{})
-		for _, config := range masterSpecs {
-			if _, ok := seen[config.AvailabilityZone]; !ok {
-				zones = append(zones, config.AvailabilityZone)
-				seen[config.AvailabilityZone] = struct{}{}
-			}
-		}
+	computeAvailabilityZones := make([]string, len(masterSpecs))
+	for i := range computeAvailabilityZones {
+		computeAvailabilityZones[i] = masterSpecs[i].AvailabilityZone
 	}
 
-	var masterRootVolumeAvailabilityZones []string
-	{
-		if defaultmpool != nil && defaultmpool.RootVolume != nil {
-			masterRootVolumeAvailabilityZones = defaultmpool.RootVolume.Zones
-		}
-		if mastermpool != nil && mastermpool.RootVolume != nil && mastermpool.RootVolume.Zones != nil {
-			masterRootVolumeAvailabilityZones = mastermpool.RootVolume.Zones
+	storageAvailabilityZones := make([]string, len(masterSpecs))
+	for i := range storageAvailabilityZones {
+		if masterSpecs[i].RootVolume != nil {
+			storageAvailabilityZones[i] = masterSpecs[i].RootVolume.Zone
```

Comment:
Again: no need to recompute storage zones. We just pick what we have in Machines.